yellowbrick/classifier/threshold.py

```diff
@@ -0,0 +1,351 @@
+            )
+        super(ThresholdVisualizer, self).__init__(model, **kwargs)
+
+        self.estimator = model
+        self.colormap = kwargs.pop('colormap', None)
+        self.color = kwargs.pop('color', None)
+        self.title = title
+        self.n_trials = n_trials
+        self.test_size_percent = test_size_percent
+        self.quantiles = quantiles
+        self.random_state = np.random.RandomState(random_state)
+
+        # to be set later
+        self.plot_data = None
+
+    def fit(self, X, y=None, **kwargs):
+        """
+        Parameters
+        ----------
+
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        kwargs: dict
+            keyword arguments passed to Scikit-Learn API.
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the visualizer
+        """
+        self.plot_data = []
+
+        for trial in range(self.n_trials):
+            train_X, test_X, train_y, test_y = train_test_split(
+                X,
+                y,
+                test_size=self.test_size_percent,
+                random_state=self.random_state)
+            self.estimator.fit(train_X, train_y)
+            # get prediction probabilities for each
+            predictions = self.estimator.predict_proba(test_X)[:, 1]
+
+            precision, recall, thresholds = precision_recall_curve(
+                test_y, predictions)
+            # add one to each so that thresh ends at 1
+            thresholds = np.append(thresholds, 1)
+            queue_rate = []
+            for threshold in thresholds:
+                queue_rate.append((predictions >= threshold).mean())
+
+            trial_data = {
+                'thresholds': thresholds,
+                'precision': precision,
+                'recall': recall,
+                'queue_rate': queue_rate
+            }
+            self.plot_data.append(trial_data)
+
+        return self
+
+    def draw(self, *kwargs):
+        """
+        Renders the visualization
+
+        Parameters
+        ----------
+        kwargs: dict
+            keyword arguments passed to Scikit-Learn API.
+
+        Returns
+        -------
+        self.ax : AxesSubplot of the visualizer
+            Returns the AxesSubplot instance of the visualizer
+        """
+        # set the colors
+        if self.colormap is not None or self.color is not None:
+            color_values = resolve_colors(
+                num_colors=3, colormap=self.colormap, color=self.color)
+        else:
+            color_values = get_color_cycle()
+
+        # unpack first three colors for visualization
+        recall_color, precision_color, queue_rate_color = color_values[:3]
+
+        if self.ax is None:
+            self.ax = plt.gca(xlim=[-1, 1], ylim=[-1, 1])
```

Comment:
This is no longer required with the new `self.ax` property in the descriptor, which automatically gets the correct axis. In fact, self.ax will never be None; so this will never get called! To ensure the limits are set correctly, set them in `finalize()`.